werkzeug/test.py

```diff
@@ -339,6 +345,19 @@ def __init__(self, path='/', base_url=None, query_string=None,
                     else:
                         self.form.setlistdefault(key).append(value)
 
+        if json:
+            if not isinstance(json, dict):
+                raise TypeError('json parameter must be of type dict')
+
+            json = _json.dumps(json)
+            json = json.encode(self.charset)
```

Comment:
No, we would have to make it conditional, depending on whether we run on Python 3. We already have a helper function for this: https://github.com/pallets/werkzeug/blob/master/werkzeug/_compat.py#L105